### PR TITLE
Add InterpolatedQuadratureParameter

### PR DESCRIPTION
**Pull request by jetuk**
A parameter that returns the interpolated quadrature of another parameter.

### CODE REVIEW

File: tests/test_parameters.py [hunk 1]
```diff
@@ -951,6 +951,31 @@ def expected_func(timestep, scenario_index):
...
+        elif i < 11:
+            value = 25 + 4*(i - 5)**2 / 2 + (i - 5) * 10
+        else:
+            value = 25 + 50 + 50 + 28 / 2 + 2
+        return value
+
+    model.run()
+
+
+
```
Comment (snorfalorpagus) [inline tests/test_parameters.py]: There should only be 2 blank lines between classes/functions.

**Comment by jetuk** [comment]
I merge master in to here so it does the new CI correctly.

**Comment by Batch21** [comment]
Gone through this and it looks good. Can't think of a use case, but is it worth making it possible to set the `a` arg of the quad function to a parameter?

**Comment by jetuk** [comment]
> ... is it worth making it possible to set the a arg of the quad function to a parameter

Probably, for completeness. I guess this should also be interpolated.

**Comment by jetuk** [comment]
@Batch21 I've addressed your comment. If you are happy could submit a review and approve?

**Pull request merged**